pyproject.toml

```diff
@@ -77,6 +77,7 @@ test = [
 ]
 type = [
   "mypy>=1.4.1,<1.5",
```

Comment:
Shall we then drop mypy from here entirely, and update here https://github.com/koxudaxi/datamodel-code-generator/blob/main/tox.ini#L64 to invoke pyright instead. And we should add a new environment to run it as part of the CI. I tend to add them as another batch of jobs to run alongside tests, see example https://github.com/tox-dev/tox/blob/main/.github/workflows/check.yaml#L58-L93 

We can then add the pkg-meta, readme and type environments there. That way all checks are done part of the CI run.